src/main/resources/avro/bdg.avdl

```diff
@@ -955,51 +670,199 @@ record VariantAnnotation {
+   Number=R are split for multi-allelic sites.
+   */
+  map<string> attributes = {};
+}
+
+/**
+ Allele encodings for genotypes.
+ */
+enum Allele {
+
+  /**
+   Reference allele.
+   */
+  REF,
+
+  /**
+   Alternate allele.
+   */
+  ALT,
+
+  /**
+   An unspecified other alternate allele. This may occur when the variant for
+   the genotype partially encoded by this allele is split as a multi-allelic site.
+   */
+  OTHER_ALT,
+
+  /**
+   Missing allele.
+   */
+  NO_CALL
+}
+
+/**
+ Genotype.
+ */
+record Genotype {
+
+  /**
+   The variant for this genotype.
+   */
+  union { null, Variant } variant = null;
+
+  /**
+   The reference contig the variant for this genotype exists on. VCF column 1 "CONTIG".
+   Used when the variant field has been left unset or set to null for performance reasons.
+   */
+  union { null, string } contigName = null;
+
+  /**
+   The zero-based start position of the variant for this genotype on the reference contig.
+   VCF column 2 "POS" converted to zero-based coordinate system, closed-open intervals.
+   Used when the variant field has been left unset or set to null for performance reasons.
+   */
+  union { null, long } start = null;
+
+  /**
+   The zero-based, exclusive end position of the variant for this genotype on the reference
+   contig. Calculated by Variant.start + Variant.referenceAllele.length(). Used when the
+   variant field has been left unset or set to null for performance reasons.
+   */
+  union { null, long } end = null;
+
+  /**
+   Sample identifier for this genotype. Join with Sample.sampleId for sample metadata.
+   VCF header line sample id.
+   */
+  union { null, string } sampleId = null;
+
+  /**
+   One or more allele values encoding this genotype. If this genotype is phased, the order of
+   the allele values is significant. VCF genotype field reserved key GT mapped to Allele enum
+   values.
+   */
+  array<Allele> alleles = [];
+
+  /**
+   True if this genotype is phased. If true, the order of alleles is significant. VCF genotype field
+   reserved key GT allele separators: '|' genotype is phased; '/' genotype is unphased.
+   */
+  union { boolean, null } phased = false;
+
+  /**
+   Phase set identifier, if any. Phased genotypes on the same contig with the same phaseSetId
+   are in the same phased set. All phased genotypes that do not have a phaseSetId are assumed to
+   belong to the same phased set. VCF genotype field reserved key PS.
+   */
+  union { null, string } phaseSetId = null;
+
+  /**
+   True if filters were applied for this variant. VCF genotype field reserved key FT any
+   value other than the missing value.
+   */
+  union { null, boolean } filtersApplied = null;
+
+  /**
+   True if all filters for this variant passed. VCF genotype field reserved key FT value PASS.
+   */
+  union { null, boolean } filtersPassed = null;
+
+  /**
+   Zero or more filters that failed for this variant. VCF genotype field reserved key FT.
```

Comment:
should "..that failed for this variant" be "..that failed for this genotype call"?
Also applies to above field.